### PR TITLE
Added tests to increase code coverage in sunpy.time package

### PR DESCRIPTION
**Pull request by m-charlton**
- code coverage increased: time.py, timerange.py
- `TimeRange.__eq__` method altered
- `TimeRange.__ne__` method added
- Minor formatting changes to satisfy pep8

#### Code coverage

- `sunpy.time.timerange`: 92% -> 96%
- `sunpy.time.time`: 58% -> 93%

#### Changes to `sunpy/time/timerange.py`

`__eq__`

Returns `NotImplemented` if an attempt is made to compare a 'TimeRange' against 
something which is *not* a `TimeRange`, in practice this evaluates to `False` see 
[here](http://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror)

I reasoned the new behaviour is *neater* than the current but, I'd very much 
appreciate feedback. 

The current behaviour is:
```
>>> dict() == TimeRange(...)
...
File ".../sunpy/time/timerange.py", line 204, in __eq__
  return ((self.start == other.start) and (self.end == other.end))
AttributeError: 'dict' object has no attribute 'start'
``` 

`__ne__`

Implemented `__ne__` as usually a good idea to define if `__eq__` is already
present. Did not define in terms of `__eq__`. See [here](http://stackoverflow.com/questions/4352244/python-should-i-implement-ne-operator-based-on-eq)
for reasoning.

Minor changes to formatting to satisfy pep8.

##### Observations:

The `TimeRange.__repr__` method has the same behaviour as the `__str__` method usually has, that is
it returns a nicely formatted string suitable for human consumption. Ideally the return 
value of `__repr__` should be able to be evaluated by `eval`. [See here](http://brennerm.github.io/posts/python-str-vs-repr.html)

If needed, I can make the changes.

### CODE REVIEW

**Comment by Cadair** [comment]
Thanks @m-charlton this looks great.

As far as `__repr__` and `__str__` go the idea of repr being able to be put through `eval` is far from as simple as it seems (for a large array it would be very challenging) and above that we tend to make `__repr__` return a nice human readable representation of the object so it's nice to see and understand when working at an interactive prompt.

That being said, I have started to make the first line of the `__repr__` methods look similar to the `object.__repr__` method, so it's obvious what type the return is and what instance it is. So if you want to make that change feel free (calling the `super` `__repr__` is a good way to do this).

**Comment by m-charlton** [comment]
That makes sense, I'll revisit the `TimeRange.__repr__` method.

Would it be possible to look at, if you not already, PR 'Add mock libs for testing, 
improve code coverage for sunpy.util.util #2001'? I'm working on some unit tests 
for `sunpy.net.helio.parser` which make extensive use of mocking, and this PR merges
the mock library for Py2.x in to mainline.

With regards to `sunpy.net.helio.parser` I think I *may* have found a subtle 
bug in the `wsdl_retriever(...)` function. Should I raise this as a separate 
issue or just mention it in the PR?

Cheers

@m-charlton

**Comment by Cadair** [comment]
I think I would put the end `>` on the end of the first line.

**Comment by m-charlton** [comment]
Will do. It will look like this:

```
>>> print(tr)
   <sunpy.time.timerange.TimeRange object at 0x101451fd0>
    Start: 2016-01-04 09:30:00
    End:   2016-06-04 09:30:00
    Center:2016-03-20 09:30:00
    Duration:152.0 days or
           3648.0 hours or
           218880.0 minutes or
           13132800.0 seconds

>>> 
```

**Comment by Cadair** [comment]
can you add a changelog entry for the `__repr__` change? (I don't think the `__ne__` exposes an API change?)

**Comment by m-charlton** [comment]
Will do.

File: sunpy/time/tests/test_time.py [hunk 1]
```diff
@@ -152,3 +168,75 @@ def test_time_string_parse_format():
         parse_time('01/06/2012')
     with pytest.raises(ValueError):
         parse_time('01/06/2012', time_string_parse_format='%d/%m/%m')
+
+    with pytest.raises(ValueError):
+        parse_time('2016', _time_string_parse_format='zz')
+
+
+def test__iter_empty():
```
Comment (nabobalis) [inline sunpy/time/tests/test_time.py]: One too many _

Comment (dpshelio) [inline sunpy/time/tests/test_time.py]: isn't the function `_iter_empty`? if so, I don't think there's many `_` :)

**Pull request merged**